File: web/src/dtos/position.ts
import { z } from "zod";
import { nameSchema, nullableUrlSchema } from "@/utils/dto";

export const AddPositionSchema = z.object({
  name: nameSchema,
  iconUrl: nullableUrlSchema,
});

export const UpdatePositionSchema = z.object({
  name: nameSchema.optional(),
  iconUrl: nullableUrlSchema.optional(),
});

export type AddPositionDTO = z.infer<typeof AddPositionSchema>;
export type UpdatePositionDTO = z.infer<typeof UpdatePositionSchema>;

export interface PositionDTO {
  positionId: number;
  presetId: number;
  name: string;
  iconUrl: string | null;
}

File: python/shared/dtos/position.py
from . import BaseDTO, BaseEntityDTO, NameStr, NullableUrlStr


class PositionDTO(BaseEntityDTO):
    position_id: int
    preset_id: int
    name: str
    icon_url: str | None


class AddPositionDTO(BaseDTO):
    name: NameStr
    icon_url: NullableUrlStr


class UpdatePositionDTO(BaseDTO):
    name: NameStr | None = None
    icon_url: NullableUrlStr = None
